File: packages/platos-client-py/platos_client/apis/agents.py
"""Agents API for platos-client."""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

from platos_client.errors import PlatosNotFoundError

if TYPE_CHECKING:
    from platos_client.client import PlatosClient, PlatosScope


class AgentsApi:
    def __init__(self, client: "PlatosClient") -> None:
        self._client = client

    async def list(self, scope: "PlatosScope | None" = None) -> list[dict[str, Any]]:
        res = await self._client._request("GET", "/api/v1/agent/agents", scope=scope)
        if not res:
            return []
        return list(res.get("agents", []))

    async def get(
        self, agent_id: str, scope: "PlatosScope | None" = None
    ) -> dict[str, Any] | None:
        try:
            return await self._client._request(
                "GET", f"/api/v1/agent/agents/{agent_id}", scope=scope
            )
        except PlatosNotFoundError:
            return None

    async def list_versions(
        self, agent_id: str, scope: "PlatosScope | None" = None
    ) -> list[dict[str, Any]]:
        res = await self._client._request(
            "GET", f"/api/v1/agent/agents/{agent_id}/versions", scope=scope
        )
        if not res:
            return []
        return list(res.get("versions", []))

File: packages/platos-client/src/apis/agents.ts
/**
 * @platosdev/client — agents API. Theme I.1.
 */

import type { PlatosClient } from "../client.js";
import { PlatosNotFoundError } from "../errors.js";
import type { PlatosAgent, PlatosScope } from "../types.js";

export class AgentsApi {
  constructor(private readonly client: PlatosClient) {}

  /**
   * List agents visible within the caller's scope. The server enforces
   * scope filtering on every query — a token from env=dev cannot see
   * env=prod agents even if it asks nicely.
   */
  async list(scope?: PlatosScope): Promise<PlatosAgent[]> {
    const res = await this.client._fetch<{ agents: PlatosAgent[] }>(
      "/api/v1/agent/agents",
      { method: "GET" },
      scope,
    );
    return res?.agents ?? [];
  }

  async get(agentId: string, scope?: PlatosScope): Promise<PlatosAgent | null> {
    try {
      return await this.client._fetch<PlatosAgent>(
        `/api/v1/agent/agents/${encodeURIComponent(agentId)}`,
        { method: "GET" },
        scope,
      );
    } catch (err) {
      if (err instanceof PlatosNotFoundError) return null;
      throw err;
    }
  }

  /** List prior versions for an agent (G.4 support). */
  async listVersions(agentId: string, scope?: PlatosScope): Promise<unknown[]> {
    const res = await this.client._fetch<{ versions: unknown[] }>(
      `/api/v1/agent/agents/${encodeURIComponent(agentId)}/versions`,
      { method: "GET" },
      scope,
    );
    return res?.versions ?? [];
  }
}
